pkg/sql/plan/build_show.go

```diff
@@ -110,10 +115,23 @@ func buildShowCreateTable(stmt *tree.ShowCreateTable, ctx CompilerContext) (*Pla
 		}
 		typ := types.Type{Oid: types.T(col.Typ.Id)}
 		typeStr := typ.String()
+		if types.IsDecimal(typ.Oid) { //after decimal fix,remove this
+			typeStr = "DECIMAL"
+		}
 		if typ.Oid == types.T_varchar || typ.Oid == types.T_char {
 			typeStr += fmt.Sprintf("(%d)", col.Typ.Width)
 		}
-		createStr += fmt.Sprintf("`%s` %s %s%s", colName, typeStr, nullOrNot, hasAttrComment)
+		updateOpt := ""
+		if col.OnUpdate != nil {
+			if f, ok := col.OnUpdate.Expr.(*plan.Expr_F); ok { //? other type
+				fStr, err := convertFunc2Str(f.F)
```

Comment:
i don't think that will be a good choice to deserialization expr.  that's more complex than your convertFunc2Str function. may be you can save the origin string to some where to avoid deserialization

Reply:
OK, I will try.